### PR TITLE
Flow timeout timing/v15

### PR DESCRIPTION
**Pull request by victorjulien**
More precise flow timeout enforcement.

Cleans up unix socket initialization.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2184

Rebase of #12084, with comments addressed.

https://redmine.openinfosecfoundation.org/issues/7455

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.flow.dhcp | 563 | 600 | 106.57% |
| | SURI_TLPR1_stats_chk | | |
| .uptime | 630 | 655 | 103.97% | 

Pipeline 23927

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.flow.dhcp | 563 | 600 | 106.57% | 

Pipeline 23944

File: src/flow-manager.c [hunk 1]
```diff
@@ -183,17 +186,30 @@ void FlowDisableFlowManagerThread(void)
...
     }
-    if (*next_ts == 0 || flow_times_out_at < *next_ts)
-        *next_ts = flow_times_out_at;
+    if (*next_ts == 0 || (uint32_t)SCTIME_SECS(timesout_at) < *next_ts)
+        *next_ts = (uint32_t)SCTIME_SECS(timesout_at);
 
-    /* do the timeout check */
-    if ((uint64_t)flow_times_out_at >= SCTIME_SECS(ts)) {
-        return false;
+    /* if time is live, we just use the tts */
```
Comment (jasonish) [inline src/flow-manager.c:201]: nit: I don't see anything named `tts` here? Perhaps something a little more descriptive, even if "thread ts".

**Review by catenacyber — APPROVED** [review]
CI : 🟢  (check rust and Debian MSRV failures unrelated)
Code : ok
Commits segmentation : ok, but
- does 584a5e5965de39658081f3eeed26e00e5b1e9b4b need its own ticket/test ?
- Can de9ceab8c784814487615ddc8558c24c36d342c2 be squashed : it looks to fix a previous commit of this PR

Commit messages : ok, but
- `fix fb ts` I do not understand
- `align struct to CLS` I do not understand

Git ID set : looks fine for me
CLA : you already contributed
Doc update : not needed
Redmine ticket : ok
Rustfmt : no rust
Tests : I think it is ok
Dependencies added: none

**Comment by jasonish** [comment]
> `fix fb ts` I do not understand

This and `tts`... Shortened terms, at least IMO without enough context surrounding their usage requiring assumed knowledge.

**Review by jasonish — APPROVED** [review]
Looks OK, but...

> `fix fb ts` I do not understand

This and `tts`... Shortened terms, at least IMO without enough context surrounding their usage requiring assumed knowledge.

**Comment by victorjulien** [comment]
Addressed your comments in #12313.

**Pull request closed without merging**